src/client/mod.rs

```diff
@@ -941,45 +943,59 @@ impl Client {
 /// // ensure a valid token is in fact valid:
 /// assert!(validate_token("Mjg4NzYwMjQxMzYzODc3ODg4.C_ikow.j3VupLBuE1QWZng3TMGH0z_UAwg").is_ok());
 ///
-/// // "cat" isn't a valid token:
-/// assert!(validate_token("cat").is_err());
-///
-/// // tokens must have three parts, separated by periods (this is still
-/// // actually an invalid token):
-/// assert!(validate_token("aaa.abcdefgh.bbb").is_ok());
-///
-/// // the second part must be _at least_ 6 characters long:
-/// assert!(validate_token("a.abcdef.b").is_ok());
-/// assert!(validate_token("a.abcde.b").is_err());
+/// // helpful to prevent typos
+/// assert!(validate_token("Njg4NzYwMjQxMzYzODc3ODg4.C_ikow.j3VupLBuE1QWZng3TMGH0z_UAwg").is_err());
 /// ```
 ///
 /// # Errors
 ///
 /// Returns a [`ClientError::InvalidToken`] when one of the above checks fail.
 /// The type of failure is not specified.
 pub fn validate_token(token: impl AsRef<str>) -> Result<()> {
-    let token = token.as_ref();
+    let (user_id, generation_time) =
+        parse_token(token.as_ref()).ok_or(Error::Client(ClientError::InvalidToken))?;
 
-    if token.is_empty() {
+    dbg!(user_id, generation_time);
```

Comment:
I don't think there is value in leaving this debug line in.

Reply:
Yes, it was an oversight on my part, sorry